### ISSUE TITLE
Custom Forerunner Scope for Scopes Framework

### ISSUE THREAD

**Issue by Scrivener07** [Scrivener07/FO4_Spartan#7]
Create a custom Forerunner scope for the [Scopes Framework](https://github.com/Scrivener07/FO4_Scopes) and [Overlays Framework](https://github.com/Scrivener07/FO4_Overlays). This scope will have a special behavior if the spartan Halo overlay is also worn. If the spartan HUD overlay is worn, then an alternative overlay texture should be loaded. If the spartan HUD overlay is not worn, then display the weapons normal scope.

To summarize, if a Halo helmet keyword is present, then use the scope w/hud. If no Halo helmet keyword is present, then use the other scope.

![forerunner_scope_wip](https://user-images.githubusercontent.com/1912967/42062167-b4b3de40-7afa-11e8-8a1e-2992201c18e4.jpg)
![forerunner_scope_withhud](https://user-images.githubusercontent.com/1912967/42062171-b69d344a-7afa-11e8-9ae1-68b644437142.jpg)
